src/main/java/core/basesyntax/service/activity/strategy/ActivityHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.activity.strategy;
+
+import core.basesyntax.model.FruitActivity;
+import core.basesyntax.service.exceptions.InvalidFruitsQuantityException;
+
+public interface ActivityHandler {
+
+    void processActivity(FruitActivity fruitActivity);
+
+    default void validateActivity(FruitActivity fruitActivity) {
+        if (fruitActivity == null) {
+            throw new NullPointerException("Cant handle null activity");
```

Comment:
to throw NPE is a bad practice. Throw RuntimeException or your own custom RuntimeException